feat: updating to include did sign (#269)
KILTprotocol/docs: docs/participate/03_treasury_proposal.md

This will help you get valuable community feedback throughout the process.
It also gives Council members an open and transparent way to measure community sentiment.

When submitting for discussion, it is best practice to include a proposal document.

> [A reference document](https://docs.google.com/document/d/1nz5GNjTL5inaI5s9djIki7DUp5zyPJV9dEI9K06kJ58/edit) can be found in the KILT Protocol discussions this can be used as a template for future proposals.

Reviewer: I think we have better documents we can refer to, don't we? Like the Dwellir proposals? Or the Galani one?

Author: I have updated with the galani one.

## DID Sign

> The proposed document MUST be DID signed.

Reviewer: I would not use spec language here, with the capital MUST stuff. Just make it lowercase, and we're good.

Author: Okie dokie

This requires a DID that MUST be associated with the proposal.
Therefore, it requires the proposer to have a DID.

> The signature provides integrity and accountability for the submitter, which can give more confidence to the community and council.

Reviewer: ```suggestion
The signature provides integrity and accountability about the submitter, which can give more confidence to the community and council.
```

> The DID Signature should be done via [DIDSign](https://didsign.io/) a dApp which is built on KILT.

Reviewer: Would remove the unnecessary details. Meaning, I would remove also the next line, about the database.

```suggestion
The DID Signature should be done via the [DIDSign](https://didsign.io/) dApp.
```

The DIDSign doesn't have a database, nor store any data by the user.

> The following guide provides a [how to create a signature using a DID](https://kilt-protocol.org/files/How-to-Guide-DIDsign.pdf) with DIDSign.

Reviewer: ```suggestion
The following guide explains [how to create a signature using a DID with DIDSign](https://kilt-protocol.org/files/How-to-Guide-DIDsign.pdf).
```

> Once the document has been signed and a link can be added to the proposal so others can verify.

Reviewer: ```suggestion
Once the document has been signed, the signature must be made available to others to verify.
The easiest way to do this is to store it somewhere publicly accessible, e.g., on IPFS or as a GitHub gist.
```

This MUST include the document and signature.

> The method of storing the document and the signature can vary, however, we recommend using IPFS or an equivelant such as Crust.

Reviewer: We don't really store the document, as the document itself can be downloaded from Google, most likely. If you want to highlight that other providers can be used, it should be explained. The way it is currently written does not make it very easy to read, at least for me.

> The document can be retrieved easily can be verified by anyone with a DID.

Reviewer: Not necessary to have a DID to verify the document. I would remove this line.

Author: Updated!

> Once the document has been signed, the signature must be made available to others to verify.
> The easiest way to do this is to store it somewhere publicly accessible, e.g., on IPFS or as a GitHub gist.
> This must include the document and signature.
> The method of storing the document and the signature can vary, however, we recommend using IPFS or an equivelant such as Crust.

Author: ```suggestion
Once the document has been signed, the signature must be made publicly available for verification.
Depending on the services and software used, accessing the storage may require different instructions, e.g., on IPFS or as a GitHub gist or cloud provider.
This must include the document and signature.
```

Author: @ntn-x2 What do you think? Has this cleared it up, or do you want to add anything else?

Reviewer: > This must include the document and signature.

This is not required. The document itself is typically hosted somewhere else than the signature. So you would download it from Google Drive, while the signature could be stored also in Google Drive, or it could be stored on IPFS.

Please include how to verify and download the necessary documents.
The following is an example done by BOTLabs GmbH.

```md

> ## How to verify the document integrity

Reviewer: I don't think this header is working as expected.

The current version of the proposal document has been digitally signed with one of the DIDs that BOTLabs GmbH controls. To verify the signature:

1. Download the PDF version of the Google Document linked above.

2. Download the DID signature of the file from IPFS, with CID QmRcYyPcCKGDQno2m5qBSZq7dftoZKuwraF9C9M96rXR36 (e.g., ipfs.io).

3. Visit didsign.io, and upload both the PDF file and the downloaded signature. The signature should verify correctly and link to the KILT tx in which the timestamp was generated.

> 4. To verify the back-link from the DID to the account to which the funds must be sent if the Proposal is accepted, copy the account in the Proposal (15oXzySe6tjF2MumHfUodH8pFQWjy2hraRmXUJXXMKKY6p3F for quick reference) and paste it in w3n.id. The account must be linked to the same web3name that signed the proposal document.

Reviewer: I would also remove this line, since we discussed the signer must not necessarily be linked to the payee.

Author: I removed the last step as it doesn't do what was intended. As we do not assume the signer and payee are the same and linked.

Reviewer: No, but we are requiring that the **submitter** is linked to the DID signing the document. So we still require a link, but not between the receiver and the DID, right?
I mean we can also not require that, but I think that's what was decided. So you can't just take someone else's proposal and put it on chain, unless you sign it yourself.

Author: The removal of the line has happened. I removed the necessary linking.

Reviewer: But we still need a link, right? Between the DID signer and the account submitter.

Author: How would we do this if someone doesn't want to link their account or they are not completely in control over it?

Reviewer: I would say that someone submitting on behalf of someone else is an edge case and not the norm. Tips are suggested by other people, but treasury proposals should be submitted by the person asking for the money. But ok, we can keep this more relaxed and add stricter requirements if we need. Let's keep it like this for now.

This requires a DID that must be associated with the proposal.
Therefore, it requires the proposer to have a DID.
The signature provides integrity and accountability about the submitter, which can give more confidence to the community and council.

The DID Signature should be done via the [DIDSign](https://didsign.io/) dApp.
The DIDSign doesn't have a database, nor store any data by the user.
The following guide explains [how to create a signature using a DID with DIDSign](https://kilt-protocol.org/files/How-to-Guide-DIDsign.pdf).

Once the document has been signed, the signature must be made publicly available for verification.
Depending on the services and software used, accessing the storage may require different instructions, e.g., on IPFS or as a GitHub gist or cloud provider.
The document should be made publicly available as well as the signature.

## Example on IPFS

An example of how to do this via IPFS using Google Drive.
Have a document ready for the proposal or discussion phase.

1. Make the document publicly available to view and download.

2. Sign the PDF version of the Document in the [DIDSign](https://didsign.io/).

3. Download the zip file and upload it to IPFS.

> 4. Following the instructions of the IPFS to pin the signature to IPFS and make publicly available. Recommendation can be [web3 storage](https://web3.storage/).

Author: ```suggestion
4. Following the instructions of the IPFS, pin the signature to IPFS and make it publicly available. One solution for doing is [web3 storage](https://web3.storage/).
```

> 5. After the documents and signature has been uploaded add the IPFS url to the proposal or discussion page for verification.

Author: ```suggestion
5. After the documents and signature have been uploaded to IPFS, add their URL to the proposal or discussion page for verification.
```

The DIDSign doesn't have a database, nor store any data by the user.
The following guide explains [how to create a signature using a DID with DIDSign](https://kilt-protocol.org/files/How-to-Guide-DIDsign.pdf).

Once the document has been signed, the signature must be made publicly available for verification.
Depending on the services and software used, accessing the storage may require different instructions, e.g., on IPFS or as a GitHub gist or cloud provider.
The document should be made publicly available as well as the signature.

## Example on IPFS

An example of how to do this via IPFS using Google Drive.
Have a document ready for the proposal or discussion phase.

1. Make the document publicly available to view and download.

2. Sign the PDF version of the Document in the [DIDSign](https://didsign.io/).

3. Download the zip file and upload it to IPFS.

> 4. Following the instructions of the IPFS to pin the signature to IPFS and make publicly available. Recommendation can be [web3 storage](https://web3.storage/).

Reviewer: It won't let me suggest, somehow.

4. Following the instructions of the IPFS, pin the signature to IPFS and make it publicly available. One solution for doing is [web3 storage](https://web3.storage/).

> 5. After the documents and signature has been uploaded add the IPFS url to the proposal or discussion page for verification.

Reviewer: 5. After the documents and signature have been uploaded to IPFS, add their URL to the proposal or discussion page for verification.
